cart/src/main/java/com/yas/cart/service/CartItemV2Service.java

```diff
@@ -46,22 +53,50 @@ public CartItemV2GetVm updateCartItem(Long productId, CartItemV2PutVm cartItemPu
         return cartItemMapper.toGetVm(savedCartItem);
     }
 
+    public List<CartItemV2GetVm> getCartItems() {
+        String currentUserId = AuthenticationUtils.extractUserId();
+        List<CartItemV2> cartItems = cartItemRepository.findByCustomerId(currentUserId);
+        return cartItemMapper.toGetVmList(cartItems);
+    }
+
+    @Transactional
+    public List<CartItemV2GetVm> deleteOrAdjustCartItem(List<CartItemV2DeleteVm> cartItemDeleteVms) {
+        validateCartItemDeleteVms(cartItemDeleteVms);
+
+        Map<Long, CartItemV2> cartItemById = getCartItemsByProductIds(cartItemDeleteVms);
+
+        List<CartItemV2> cartItemsToDelete = new ArrayList<>();
+        List<CartItemV2> cartItemsToAdjust = new ArrayList<>();
+
+        for (CartItemV2DeleteVm cartItemDeleteVm : cartItemDeleteVms) {
+            CartItemV2 cartItem = cartItemById.get(cartItemDeleteVm.productId());
+            if (cartItem == null) {
+                continue;
+            }
+
+            if (cartItem.getQuantity() <= cartItemDeleteVm.quantity()) {
+                cartItemsToDelete.add(cartItem);
+            } else {
+                cartItem.setQuantity(cartItem.getQuantity() - cartItemDeleteVm.quantity());
+                cartItemsToAdjust.add(cartItem);
+            }
```

Comment:
should use optional to check null

Reply:
Hi, I’ve refactored the code as you suggested using Optional. Here’s the updated snippet:

```java
            Optional<CartItemV2> optionalCartItem = Optional.ofNullable(cartItemById.get(cartItemDeleteVm.productId()));
            optionalCartItem.ifPresent(cartItem -> {
                if (cartItem.getQuantity() <= cartItemDeleteVm.quantity()) {
                    cartItemsToDelete.add(cartItem);
                } else {
                    cartItem.setQuantity(cartItem.getQuantity() - cartItemDeleteVm.quantity());
                    cartItemsToAdjust.add(cartItem);
                }
            });
```

Does this implementation align with your suggestion? It would be great if you could provide some additional examples! Thank you very much.

Reply:
Yes, I think it looks good.